### ISSUE TITLE
Error: "window is not defined" on server seide rendering

### ISSUE THREAD

**Issue by letsspeak** [gregnb/mui-datatables#398]
Thanks for your great job!!

# Summary

Error: "window is not defined" on server seide rendering next.js app.

# Enviroments

```
"react": "^16.6.3",
"next": "^7.0.2",
"@material-ui/core": "^3.7.0",
"mui-datatables": "^2.0.0-beta-53",
```

# Error detail

```
window is not defined
ReferenceError: window is not defined
    at new t (/path/to/node_modules/mui-datatables/src/MUIDataTable.js:154:27)
    at processChild (/path/to/node_modules/react-dom/cjs/react-dom-server.node.development.js:2750:14)
    at resolve (/path/to/node_modules/react-dom/cjs/react-dom-server.node.development.js:2716:5)
    at ReactDOMServerRenderer.render (/path/to/node_modules/react-dom/cjs/react-dom-server.node.development.js:3100:22)
    at ReactDOMServerRenderer.read (/path/to/node_modules/react-dom/cjs/react-dom-server.node.development.js:3059:29)
    at renderToString (/path/to/node_modules/react-dom/cjs/react-dom-server.node.development.js:3526:27)
    at Object.renderPage (/path/to/node_modules/next/dist/server/render.js:319:26)
    at Function.MyDocument.getInitialProps (webpack:///pages/_document.js:65:1)
    at _callee$ (/path/to/node_modules/next/dist/lib/utils.js:86:30)
    at tryCatch (/path/to/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/path/to/node_modules/regenerator-runtime/runtime.js:288:22)
    at Generator.prototype.(anonymous function) [as next] (/path/to/node_modules/regenerator-runtime/runtime.js:114:21)
    at asyncGeneratorStep (/path/to/node_modules/@babel/runtime-corejs2/helpers/asyncToGenerator.js:5:24)
    at _next (/path/to/node_modules/@babel/runtime-corejs2/helpers/asyncToGenerator.js:27:9)
    at /path/to/node_modules/@babel/runtime-corejs2/helpers/asyncToGenerator.js:34:7
    at new Promise (<anonymous>)
    at new F (/path/to/node_modules/core-js/library/modules/_export.js:36:28)
    at /path/to/node_modules/@babel/runtime-corejs2/helpers/asyncToGenerator.js:23:12
    at loadGetInitialProps (/path/to/node_modules/next/dist/lib/utils.js:49:31)
    at _callee3$ (/path/to/node_modules/next/dist/server/render.js:337:51)
    at tryCatch (/path/to/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/path/to/node_modules/regenerator-runtime/runtime.js:288:22)
```

# How to avoid this error

I switched to previous version `mui-datatables@2.0.0-beta-52` and it works!

```
yarn remove mui-datatables
yarn add mui-datatables@2.0.0-beta-52
```

# Cause

Perhaps following change 6 days ago is causing this error.
https://github.com/gregnb/mui-datatables/pull/388/files#diff-4f4aff53b74d2c551e37786d32b82e3aR155

**Comment by gregnb** [comment]
this should be resolved upgrade to the latest version

**Comment by mpareja** [comment]
The `window` reference still results in a ReferenceError on version `2.0.0-beta-54`.

**Comment by gregnb** [comment]
@mpareja Do you want to send in a PR?

**Comment by mpareja** [comment]
Hi @gregnb & @letsspeak,

We _could_ fix this using `if (typeof window !== 'undefined')`; however, I'm not sure mui-datatables _ought_ to be setting that global variable on behalf of applications to begin with.

I looked into `__MUI_USE_NEXT_TYPOGRAPHY_VARIANTS__` a bit more. The intent is a bit confusing, but I believe we don't actually need to set it to `true`. The version of material-ui listed in the peer dependencies works with the new variants without setting this field. I believe the intended use of this flag is for applications that want to start receiving warnings about the use of deprecated variants so that they can make a complete migration. Applications can opt-in to turn on the warnings by setting that variable or passing `useNextVariants = true` to `createMuiTheme`.

In other words, can you test out if the removal of setting that flag breaks things for you? Everything seems to work fine for me using `@material-ui/core@3.9.2`. I don't have a test-bed for using `@material-ui/core@3.2.0`, though.

**Comment by gregnb** [comment]
@mpareja Thanks for the information! It is a bit confusing and lately I've had maybe an hour a week to spend on issues so I took a quick look at their page and applied that fix.  Like you mentioned if users come back with issues they should `useNextVariants` from their own theme provider.

If you want to send in a PR with the removal I'll accept it

**Comment by mpareja** [comment]
Okay, sounds good. I'll put something together.
